File: LeetCode/Simulation/Golang/Leetcode1710-maximum-units-on-a-truck.go
package sbgo

import "sort"

func maximumUnits(boxTypes [][]int, truckSize int) int {
	sort.Slice(boxTypes, func(i, j int) bool {
		return boxTypes[i][1] > boxTypes[j][1]
	})
	ans := 0
	for _, x := range boxTypes {
		if x[0] >= truckSize {
			ans += truckSize * x[1]
			break
		}
		ans += x[0] * x[1]
		truckSize -= x[0]
	}
	return ans
}

File: LeetCode/Simulation/JS/Leetcode1710-maximum-units-on-a-truck.js
var maximumUnits = function (boxTypes, truckSize) {
  boxTypes.sort((a, b) => b[1] - a[1]);
  let ans = 0;
  for (const x of boxTypes) {
    if (x[0] >= truckSize) {
      ans += truckSize * x[1];
      break;
    }
    ans += x[0] * x[1];
    truckSize -= x[0];
  }
  return ans;
};